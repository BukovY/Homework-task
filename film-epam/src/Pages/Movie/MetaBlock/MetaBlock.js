import React from "react";
import './MetaBlock.css'

const MetaBlock = ({title, meta, prefix}) => {

    return(<div>
        <p className='meta_title'>{title}:</p>
        <p>{prefix? prefix: ''}{meta}</p>
    </div>)
}
export default MetaBlock