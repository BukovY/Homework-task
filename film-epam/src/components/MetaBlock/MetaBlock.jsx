import React from "react";
import s from "./MetaBlock.module.sass";

const MetaBlock = ({ title, meta, prefix }) => {
  return (
    <div>
      <p className={s.meta_title}>{title}:</p>
      <p>
        {prefix ? prefix : ""}
        {meta}
      </p>
    </div>
  );
};
export default MetaBlock;
