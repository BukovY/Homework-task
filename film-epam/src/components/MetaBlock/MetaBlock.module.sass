.hide
  display: none

.padding
  padding-bottom: 10px