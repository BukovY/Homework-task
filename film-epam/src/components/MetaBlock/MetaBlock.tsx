import React, { FC } from "react";
import styles from "./MetaBlock.module.sass";
import classNames from "classnames/bind";
import Typography from "@material-ui/core/Typography";
type MetaBlockProps = {
  title: string;
  meta: string;
  prefix: string;
};
export const MetaBlock: FC<MetaBlockProps> = ({ title, meta, prefix }) => {
  const cx = classNames.bind(styles);
  const metaBlockClass = cx("padding", {
    hide: !meta,
  });
  return (
    <div className={metaBlockClass}>
      <Typography variant="body2">{title}:</Typography>
      <Typography variant="body1">
        {prefix}
        {meta}
      </Typography>
    </div>
  );
};
