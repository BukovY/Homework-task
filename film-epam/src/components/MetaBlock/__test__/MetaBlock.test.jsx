import React from "react";
import rerender from "react-test-renderer";
import { MetaBlock } from "../MetaBlock";

describe("Testing <MetaBlock/> Component", () => {
  it("Metcblock view correct", () => {
    const tree = rerender
      .create(
        <MetaBlock meta={"meta text"} prefix={"prefix"} title={"title"} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
