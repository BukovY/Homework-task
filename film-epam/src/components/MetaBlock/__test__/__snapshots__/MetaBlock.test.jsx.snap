// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <MetaBlock/> Component Metcblock view correct 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 makeStyles-padding-2"
>
  <p
    className="MuiTypography-root MuiTypography-body2"
  >
    title
    :
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    prefix
    meta text
  </p>
</div>
`;
